shared/src/question-generators/time/utilsStars/utilsWhile.ts

```diff
@@ -0,0 +1,271 @@
+import {
+  BoundsOptions,
+  createIfCondition,
+  createWhileChangeValues,
+  createWhileLine,
+  IfOptions,
+} from "@shared/question-generators/time/utils.ts"
+import { calculateNumberOfStars } from "@shared/question-generators/time/utilsStars/calculateNumberOfStars.ts"
+import { generateDifferentAmountOfStarPrints } from "@shared/question-generators/time/utilsStars/utils.ts"
+import { conditionCheck } from "@shared/question-generators/time/utilsStars/utilsWhileCondition.ts"
+import { computeStartEndVarsWhile } from "@shared/question-generators/time/utilsStars/utilsWhileVarValues.ts"
+import {
+  printStarsNew,
+  PseudoCode,
+  PseudoCodeBlock,
+  PseudoCodeWhile,
+} from "@shared/utils/pseudoCodeUtils.ts"
+import Random from "@shared/utils/random.ts"
+
+/**
+ *
+ * You have many different cases how the loop is constructed
+ *
+ * The options are listed below
+ *
+ * Example result:
+ * let t = 8, let n = 8
+ * while t == n:
+ *   if t is odd
+ *     print("*")
+ *   t = floor(t / 2)
+ *   n += 3
+ *
+ * @param firstVariableName the name of the first variable
+ * @param secondVariableName second variable
+ * @param random
+ */
+export function createWhileLoop(firstVariableName: string, secondVariableName: string, random: Random) {
+  const { numPrint, numPrintElse, numPrintAfterAll } = generateDifferentAmountOfStarPrints(random)
+
+  const {
+    firstVariableValue,
+    secondVariableValue,
+    endValue,
+    variableOrder,
+    compareOption,
+    endManipulation,
+    whileCaseOption,
+  } = computeStartEndVarsWhile(random)
+
+  if (firstVariableValue === -10 && secondVariableValue === -10) {
```

Comment:
I think it's better to use the initial value `undefined` instead of -10 to indicate that the variable value has not been set yet.